src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -40,416 +44,554 @@
-						
-			if (log.isDebugEnabled()) log.debug ("The status line returned: "+statusLine);
-			
-			int statusCode = statusLine.getStatusCode();
-			socket = connectResponse.getSocket();
-			if ( socket != null && statusCode == HttpStatus.SC_OK ) {				
-				//we have a socket and a 200 status. 
-				//Could still be a false positive though, if the server ignored the method, 
-				//and did not recognise the URL, so redirected to a login page, for instance
-				//Remediation: Check the contents match the expected third party contents.
-				if (log.isDebugEnabled()) log.debug("Raw Socket established, in theory to "+thirdpartyHost);
-				
-				OutputStream os = socket.getOutputStream();
-				InputStream is = socket.getInputStream();
-				
-				PrintWriter pw = new PrintWriter(os, false);
-				pw.write("GET http://"+thirdpartyHost + ":"+thirdpartyPort + "/ HTTP/1.1\n");
-				pw.write("Host: "+thirdpartyHost+"\n\n");
-				pw.flush();
-				
-				//read the response via a 4k buffer
-				ByteArrayOutputStream bos = new ByteArrayOutputStream ();
-				byte [] buffer = new byte [1024*4];
-				int bytesRead = is.read(buffer);
-				int totalBytesRead = 0;
-				while (bytesRead > -1) {
-					totalBytesRead+= bytesRead;					
-					bos.write(buffer, 0, bytesRead);					
-					bytesRead = is.read(buffer);
-				}
-				String response = new String (bos.toByteArray());				
-				if (log.isDebugEnabled()) log.debug("Response is "+totalBytesRead+" bytes: \n"+response);
-				
-				Matcher m = thirdPartyContentPattern.matcher(response);
-				if (m.matches()) {
-					if (log.isDebugEnabled()) log.debug("Response matches expected third party pattern!");
-					is.close();
-					os.close();
-					bos.close();
-					socket.close();
-					return true;
-				} else { 
-					if (log.isDebugEnabled()) log.debug("Response does *not* match expected third party pattern");
-				}
-				is.close();
-				os.close();
-				bos.close();
-				socket.close();
-				return false;				
-			}
-			else  {
-				if (log.isDebugEnabled()) { 
-					log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
-					log.debug ("The status line returned: "+statusLine);
-				}
-				return false;
-			} 
-		}
-		catch (Exception e) {
-			if (log.isDebugEnabled()) log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
-		}
-		return false;
-	}
+    /* WebDAV methods */
+    private static final String COPY = "COPY";
+    private static final String LOCK = "LOCK";
+    private static final String MKCOL = "MKCOL";
+    private static final String MOVE = "MOVE";
+    private static final String PROPFIND = "PROPFIND";
+    private static final String PROPPATCH = "PROPPATCH";
+    private static final String UNLOCK = "UNLOCK";
+
+    /* These are the 'default' HTTP methods which are considered as insecure */
+    public static final String[] INSECURE_DEFAULT_METHODS = {
```

Comment:
- `private`.
 - This could be a `List`.
 - I'd keep the old set of methods separated, better to not mix the ones unsafe with others that aren't (some of them are in fact expected to be implemented/accessible, sometimes).